lib/driver.rb

```diff
@@ -0,0 +1,53 @@
+require_relative "csv_record"
+
+module RideShare
+  class Driver < CsvRecord
+    attr_reader :name, :vin, :trips
+    attr_accessor :status
+
+    def initialize(id:, name:, vin:, status: :AVAILABLE, trips: nil)
+      super(id)
+      @name = name
+      @vin = vin.length == 17 ? vin : (raise ArgumentError.new("Invalid Vin: #{vin.length} length"))
+      @status = status
+      if ![:AVAILABLE, :UNAVAILABLE].include? (status)
+        raise ArgumentError, "Invalid Status"
+      end
+
+      @trips = trips || []
+    end
+
+    def add_trip(trip)
+      @trips << trip
+    end
+
+    def average_rating(completed_trips = non_inprogress_trips)
+      average_rating = completed_trips.sum do |trip|
+        trip.rating
+      end
+      average_rating /= completed_trips.length.to_f if completed_trips.length > 0
```

Comment:
I like how you took advantage of the fact that the sum of an empty array is `0` here.  

One small thing: since you return `average_rating` on the very next line I'd probably just directly return it here.
```suggestion
      return average_rating / completed_trips.length.to_f if completed_trips.length > 0
```